File: Top-Interview-Questions-Leetcode/Easy-Collections/other/hamming-distance.cpp
class Solution {
public:
    int hammingDistance(int x, int y) {
        int cnt = 0;
        
        while (x or y) {
            cnt += ((x & 1) ^ (y & 1));
            x >>= 1;
            y >>= 1;
        }
        
        return cnt;
    }
};

File: Top-Interview-Questions-Leetcode/Easy-Collections/other/hamming-distance.py
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        cnt = 0
        
        while x != 0 or y != 0:
            cnt += ((x & 1) ^ (y & 1))
            x >>= 1
            y >>= 1
        
        return cnt